### PR TITLE
[NFC][SYCL] Remove deprecated spelling of IntelReqdSubGroupSize attribute

### PR DESCRIPTION
**Pull request by smanna12**
Commit b2da2c8 added support for new spelling [[intel::reqd_sub_group_size()]]
and deprecated previous spelling [[cl::intel_reqd_sub_group_size()]] of the
IntelReqdSubGroupSize attribute to match with spec.

This patch removes that deprecated spelling and updates several tests
of IntelReqdSubGroupSize attribute with new spelling.

Signed-off-by: Soumi Manna <soumi.manna@intel.com>

### CODE REVIEW

File: sycl/test/inline-asm/asm_16_empty.cpp [hunk 1]
```diff
@@ -19,7 +19,8 @@ struct KernelFunctor : WithOutputBuffer<T> {
   void operator()(cl::sycl::handler &cgh) {
     auto C = this->getOutputBuffer().template get_access<cl::sycl::access::mode::write>(cgh);
     cgh.parallel_for<KernelFunctor<T>>(
-        cl::sycl::range<1>{this->getOutputBufferSize()}, [=](cl::sycl::id<1> wiID) [[cl::intel_reqd_sub_group_size(16)]] {
+        cl::sycl::range<1>{this->getOutputBufferSize()}, [=
+    ](cl::sycl::id<1> wiID) [[intel::reqd_sub_group_size(16)]] {
```
Comment (alexbatashev) [inline sycl/test/inline-asm/asm_16_empty.cpp:23]: clang-format does bad job on this constructs. I'd prefer it to be like this:
```suggestion
    // clang-format off
        cl::sycl::range<1>{this->getOutputBufferSize()}, 
    [=](cl::sycl::id<1> wiID) [[intel::reqd_sub_group_size(16)]] {
      // clang-format on
```

If you agree, please apply to all such cases.

File: sycl/test/inline-asm/asm_no_operands.cpp [hunk 2]
```diff
@@ -25,10 +25,11 @@ int main() {
...
     cgh.parallel_for<no_operands_kernel>(
-        NumOfWorkItems, [=](cl::sycl::id<1> WIid) [[cl::intel_reqd_sub_group_size(8)]] {
+        NumOfWorkItems, [=](cl::sycl::id<1> WIid)
+                            [[intel::reqd_sub_group_size(8)]] {
 #if defined(INLINE_ASM) && defined(__SYCL_DEVICE_ONLY__)
-          asm("barrier");
+                              asm("barrier");
 #endif
-        });
+                            });
```
Comment (alexbatashev) [inline sycl/test/inline-asm/asm_no_operands.cpp:33]: These changes seem to be unrelated.

File: sycl/test/inline-asm/letter_example.cpp [hunk 3]
```diff
@@ -29,7 +29,8 @@ int main() {
   q.submit([&](cl::sycl::handler &cgh) {
      cgh.parallel_for<kernel_name>(
          cl::sycl::range<1>(problem_size), [=](cl::sycl::id<1> idx)
-                                               [[cl::intel_reqd_sub_group_size(16)]] {
+                                               [[intel::reqd_sub_group_size(
+                                                   16)]] {
```
Comment (alexbatashev) [inline sycl/test/inline-asm/letter_example.cpp:33]:
```suggestion
                                               // clang-format off
                                               [[intel::reqd_sub_group_size(16)]] {
                                                 // clang-format on
```
Same as above.

File: sycl/test/inline-asm/malloc_shared_32.cpp [hunk 4]
```diff
@@ -58,12 +57,12 @@ int main() {
         svm_scatter.4.1 (M1, 16) %1.0 V53.0
     }
     )a" ::"rw"(&b[i]),
-                            "rw"(&b[i] + 16), "rw"(&a[i]), "rw"(&a[i] + 16), "rw"(&c[i]),
-                            "rw"(&c[i] + 16));
+                        "rw"(&b[i] + 16), "rw"(&a[i]), "rw"(&a[i] + 16),
+                        "rw"(&c[i]), "rw"(&c[i] + 16));
```
Comment (alexbatashev) [inline sycl/test/inline-asm/malloc_shared_32.cpp:61]: These changes are also irrelevant.

**Review by alexbatashev — APPROVED** [review]
`sycl/` contents LGTM

**Review by Fznamznon — APPROVED** [review]
FE changes look fine

**Comment by bader** [comment]
@smanna12, could you address the comments from @alexbatashev, please?

File: sycl/test/inline-asm/asm_16_empty.cpp (hunk 1 above)
Comment (smanna12) [reply sycl/test/inline-asm/asm_16_empty.cpp:23]: Yes, I agree. I will update all cases.

File: sycl/test/inline-asm/letter_example.cpp (hunk 3 above)
Comment (smanna12) [reply sycl/test/inline-asm/letter_example.cpp:33]: I will update this too.

File: sycl/test/inline-asm/malloc_shared_32.cpp (hunk 4 above)
Comment (smanna12) [reply sycl/test/inline-asm/malloc_shared_32.cpp:61]: Yes, this is unrelated. Clang-format complained about this.  I will change this to the original.

File: sycl/test/inline-asm/asm_no_operands.cpp (hunk 2 above)
Comment (smanna12) [reply sycl/test/inline-asm/asm_no_operands.cpp:33]: Yes, this is unrelated. Clang-format complained about this.  I will change this to the original.

**Review by premanandrao — DISMISSED** [review]
LGTM

**Review by smanna12 — COMMENTED** [review]
Clang-format makes unrelated changes here. I have turned the format off.

File: sycl/test/inline-asm/letter_example.cpp [hunk 5]
```diff
@@ -39,7 +41,9 @@ int main() {
                                                               :
                                                               : "rw"(&a[i]));
 #else
+           // clang-format off
                                                  a[idx[0]]++;
+      // clang-format on
 #endif
                                                });
```
Comment (smanna12) [inline sycl/test/inline-asm/letter_example.cpp:48]: Clang-format makes unrelated change here. I have turned the format off.

File: sycl/test/inline-asm/malloc_shared_32.cpp [hunk 6]
```diff
@@ -58,6 +59,7 @@ int main() {
         svm_scatter.4.1 (M1, 16) %1.0 V53.0
     }
     )a" ::"rw"(&b[i]),
+                        // clang-format off
                             "rw"(&b[i] + 16), "rw"(&a[i]), "rw"(&a[i] + 16), "rw"(&c[i]),
                             "rw"(&c[i] + 16));
 #else
```
Comment (smanna12) [inline sycl/test/inline-asm/malloc_shared_32.cpp:65]: Clang-format makes unrelated change here. I have turned the format off.

**Review by alexbatashev — APPROVED** [review]

**Comment by smanna12** [comment]
Thanks everyone for the review.
